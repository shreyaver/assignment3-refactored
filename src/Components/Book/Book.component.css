.book {
    margin-top: 3%; 
    margin-bottom: 1.5%;
    width: 300px;
    min-height: 400px;
    display: flex;
    flex-direction: column;
    background-repeat: no-repeat;
    background-size: contain;
}
.imagePanel {
    flex: 1;
}
.bookData {
    font-size: 1.25rem;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
    padding: 0px 10px 10px 15px;
}
.bookData .bookName {
    align-self: center;
}
.bookData span {
    font-size: 1.85rem;
    font-weight: bold;
}
.likedHeartIcon, .heartIcon {
    height: 30px;
    width: 30px;
    border: none;
    float: right;
    background-color: transparent;
    outline: none;
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
}
.bookName {
    text-align: center;
    text-transform: uppercase;
}
.heartIcon {
    background-image: url("../../Icons/heart-black.svg");

}
.likedHeartIcon {
    background-image: url("../../Icons/heart-red.svg");
}
