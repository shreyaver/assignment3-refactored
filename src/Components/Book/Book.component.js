import React from 'react';
import './Book.component.css';
import PropTypes from 'prop-types';
import coverImage from '../../Images/cover_image.jpeg';
import roundOffRating from '../../Helpers/roundOffRating/roundOffRating';

const Book = (props) => {
  const { book, handleLikeClick } = props;
  const likeButton = book.liked === true ? 'likedHeartIcon' : 'heartIcon';
  return (
    <div className="book" style={{ backgroundImage: `url(${coverImage})` }}>
      <div className="imagePanel" />
      <div className="bookData">
        <p className="bookName">{book.Name}</p>
        <span>
          {roundOffRating(book.rating)}
          <button type="button" className={likeButton} onClick={handleLikeClick(book)} />
        </span>
      </div>
    </div>
  );
};
Book.propTypes = {
  book: PropTypes.shape({
    Name: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    Author: PropTypes.string,
    id: PropTypes.number,
  }).isRequired,
  handleLikeClick: PropTypes.func.isRequired,
};
export default Book;
