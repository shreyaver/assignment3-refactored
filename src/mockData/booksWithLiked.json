{ "books": {
  "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62,
        "liked": false
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54,
      "liked": false
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54,
      "liked": false
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93,
      "liked": false
     }]
}}