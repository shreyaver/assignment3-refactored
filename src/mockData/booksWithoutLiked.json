{ "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93
     }]
}