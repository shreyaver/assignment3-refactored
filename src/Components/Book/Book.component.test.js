import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Book from './Book.component';
import mockData from '../../mockData/booksWithLiked.json';

describe('Book', () => {
  const bookData = Object.values(mockData.books)[0][0];
  it('renders without crashing', () => {
    const book = renderer.create(<Book book={bookData} handleLikeClick={jest.fn()} />).toJSON();
    expect(book).toMatchSnapshot();
  });
  it('sets correct class when book is liked', () => {
    const wrapper = shallow(<Book book={Object.assign({}, bookData, { liked: true })} handleLikeClick={jest.fn()} />);
    expect(wrapper.exists('.likedHeartIcon')).toEqual(true);
    expect(wrapper.exists('.heartIcon')).toEqual(false);
  });
  it('sets correct class when book is not liked', () => {
    const wrapper = shallow(<Book book={Object.assign({}, bookData, { liked: false })} handleLikeClick={jest.fn()} />);
    expect(wrapper.exists('.heartIcon')).toEqual(true);
    expect(wrapper.exists('.likedHeartIcon')).toEqual(false);
  });
  it('calls function passed as prop when like button is clicked', () => {
    const mockFunction = jest.fn();
    const mockHandlelike = () => mockFunction;
    const wrapper = shallow(<Book book={Object.assign({}, bookData, { liked: false })} handleLikeClick={mockHandlelike} />);
    expect(mockFunction).not.toHaveBeenCalled();
    wrapper.find('.heartIcon').simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});
