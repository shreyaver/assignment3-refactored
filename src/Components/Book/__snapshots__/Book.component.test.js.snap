// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders without crashing 1`] = `
<div
  className="book"
  style={
    Object {
      "backgroundImage": "url(cover_image.jpeg)",
    }
  }
>
  <div
    className="imagePanel"
  />
  <div
    className="bookData"
  >
    <p
      className="bookName"
    >
      Harry Potter and the Deathly Hallows (Harry Potter, #7)
    </p>
    <span>
      4.6
      <button
        className="heartIcon"
        type="button"
      />
    </span>
  </div>
</div>
`;
